packages/sui-lint/stylelint.config.js

```diff
@@ -0,0 +1,65 @@
+function suitSelector(componentName) {
+  const ns = '[a-z]{2,3}'
+  const WORD = '[a-z0-9][a-zA-Z0-9]*'
+  const descendant = `(?:-${WORD})?`
+  const modifier = `(?:--${WORD}(?:\\.${ns}${componentName}${descendant}--${WORD})*)?`
+  const attribute = '(?:\\[.+\\])?'
+  const state = `(?:\\.(is|has)-${WORD})*`
+  const body = descendant + modifier + attribute + state
+  return new RegExp(`^\\.${ns}${componentName}\\b${body}$`)
+}
+
+module.exports = {
+  plugins: [
+    'stylelint-selector-bem-pattern',
+    'stylelint-scss',
+    'stylelint-prettier'
+  ],
+  rules: {
+    'prettier/prettier': [
+      true,
+      {
+        printWidth: 80,
+        tabWidth: 2,
+        singleQuote: true,
+        bracketSpacing: false,
+        useTabs: false
+      }
+    ],
+    'plugin/selector-bem-pattern': {
+      componentName: /^[a-zA-Z]+$/,
+      componentSelectors: suitSelector
+    },
+    'scss/at-rule-no-unknown': true,
+    'at-rule-no-unknown': null,
+    'block-no-empty': true,
+    'color-no-invalid-hex': true,
+    'comment-no-empty': true,
+    'declaration-block-no-duplicate-properties': [
+      true,
+      {
+        ignore: ['consecutive-duplicates-with-different-values'] // useful for fallback properties
+      }
+    ],
+    'declaration-block-no-shorthand-property-overrides': true,
```

Comment:
Just wondering if all these `stylelint` rules and the previous ones defined by `sass-lint' are going to have quite differences between them..

Reply:
*Tried* to keep exactly the same rules plus added some recommended ones. Some are just breaking, like the zero leading, as the stylelint-prettier is enforcing it (and is not configurable).